waltz-data/src/main/java/com/khartec/waltz/data/application/ApplicationIdSelectorFactory.java

```diff
@@ -205,4 +217,21 @@ private String findEmployeeId(EntityReference ref) {
                 .where(dsl.renderInlined(appCapability.CAPABILITY_ID.in(capabilitySelector)));
     }
 
+
+    private Select<Record1<Long>> mkForDataType(EntityReference ref, HierarchyQueryScope scope) {
+        ImmutableIdSelectionOptions dtSelectorOptions = ImmutableIdSelectionOptions.builder()
+                .entityReference(ref)
+                .scope(scope)
+                .build();
+
+        Select<Record1<Long>> dtSelector = dataTypeIdSelectorFactory.apply(dtSelectorOptions);
+
+        return dsl
+                .selectDistinct(dataTypeUsage.ENTITY_ID)
+                .from(dataTypeUsage)
+                .join(dataType).on(dataType.CODE.eq(dataTypeUsage.DATA_TYPE_CODE))
+                .where(dsl.renderInlined(dataTypeUsage.ENTITY_KIND.eq(EntityKind.APPLICATION.name())
```

Comment:
Good suggestion. Will do it in the morning.

On 9 Aug 2016 18:32, "David Watkins" <notifications@github.com> wrote:

> In waltz-data/src/main/java/com/khartec/waltz/data/application/
> ApplicationIdSelectorFactory.java
> <https://github.com/khartec/waltz/pull/414#discussion_r74105415>:
>
> > @@ -205,4 +217,21 @@ private String findEmployeeId(EntityReference ref) {
> >                  .where(dsl.renderInlined(appCapability.CAPABILITY_ID.in(capabilitySelector)));
> >      }
> >
> > +
> > +    private Select<Record1<Long>> mkForDataType(EntityReference ref, HierarchyQueryScope scope) {
> > +        ImmutableIdSelectionOptions dtSelectorOptions = ImmutableIdSelectionOptions.builder()
> > +                .entityReference(ref)
> > +                .scope(scope)
> > +                .build();
> > +
> > +        Select<Record1<Long>> dtSelector = dataTypeIdSelectorFactory.apply(dtSelectorOptions);
> > +
> > +        return dsl
> > +                .selectDistinct(dataTypeUsage.ENTITY_ID)
> > +                .from(dataTypeUsage)
> > +                .join(dataType).on(dataType.CODE.eq(dataTypeUsage.DATA_TYPE_CODE))
> > +                .where(dsl.renderInlined(dataTypeUsage.ENTITY_KIND.eq(EntityKind.APPLICATION.name())
>
> Perhaps pull the condition out so the code will be more like:
>
>
> Condition condition = dataTypeUsage.ENTITY_KIND.eq(EntityKind.APPLICATION.name())
>     .and(....);
>
> ....
> dsl.select(....)
>     ...
>     .where(dsl.renderInlined(condition))
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/khartec/waltz/pull/414/files/7ee6b61311f55943bb2ea3c6b92851fc8b8e11f1#r74105415>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC452_vH2eYByyf087FTdYni1rS-b1hPks5qeLm9gaJpZM4JgTQJ>
> .
>